### PR TITLE
ref(device-contexts): Add function that returns a more detailed desc of a language

### LINKED ISSUE #26004: Display language of Device context in a more human readable form
## Summary

The [device context](https://develop.sentry.dev/sdk/event-payloads/contexts/#device-context) of an event has a field
for the language. It's most like in one of these two letter language format, like `en_US`.
I would like to see a more human readable description of it. For example `en_US` should be `English (United States)`

## Motivation

Having just the two letter code does not really help me that much as I would always have to look up which language that is.
Having a more human readable description would remove this step.

## Additional Context

None

### PR DESCRIPTION
**Pull request by priscilawebdev**
closes https://github.com/getsentry/sentry/issues/26004

### CODE REVIEW

File: static/app/components/events/contexts/utils.tsx [hunk 1]
```diff
@@ -63,6 +65,39 @@ export function getRelativeTimeFromEventDateCreated(
...
+      });
+
+      const countryName = regionNames.of(countryAbbreviation.toUpperCase());
+
+      return `${languageName} (${countryName})`;
+    }
+
+    return languageName;
+  } catch (error) {
+    Sentry.captureException(error);
```
Comment (scttcper) [inline static/app/components/events/contexts/utils.tsx:96]: I'm not familiar with how many browsers have `Intl.DisplayNames` should we check for it first before logging an exception

Comment (scttcper) [reply static/app/components/events/contexts/utils.tsx:96]: nevermind seems okay https://caniuse.com/mdn-javascript_builtins_intl_displaynames_displaynames

**Review by scttcper — APPROVED** [review]

File: static/app/components/events/contexts/utils.tsx [hunk 2]
```diff
@@ -63,6 +65,39 @@ export function getRelativeTimeFromEventDateCreated(
...
 }
 
+// Typescript doesn't have types for DisplayNames yet and that's why the type assertion "any" is needed below.
+// There is currently an open PR that intends to introduce the types https://github.com/microsoft/TypeScript/pull/44022
+export function getFullLanguageDescription(locale: string) {
+  const sentryAppLanguageCode = ConfigStore.get('languageCode');
+
+  const [languageAbbreviation, countryAbbreviation] = locale.includes('_')
+    ? locale.split('_')
+    : locale.split('-');
```
Comment (BYK) [inline static/app/components/events/contexts/utils.tsx:75]: If you ever get a name like `a-b-c`, you're gonna be in trouble. Could that happen?

File: static/app/components/events/contexts/utils.tsx [hunk 3]
```diff
@@ -63,6 +65,39 @@ export function getRelativeTimeFromEventDateCreated(
...
+
+  try {
+    const languageNames = new (Intl as any).DisplayNames(sentryAppLanguageCode, {
```
Comment (BYK) [inline static/app/components/events/contexts/utils.tsx:78]: Might be good to comment here about the `as any` cast: https://github.com/microsoft/TypeScript/issues/41338

**Pull request merged**